app/audit/AuditDetails.scala

```diff
@@ -16,16 +16,54 @@
 
 package audit
 
+import cats.data.Ior
+import config.Constants
 import models.ChannelType
+import models.EORINumber
+import models.TURN
 import play.api.libs.json.JsObject
-import play.api.libs.json.OWrites
 import play.api.libs.json.Json
+import play.api.libs.json.OWrites
+
+sealed abstract class AuditDetails {
+  def channel: ChannelType
+  def customerId: String
+  def json: JsObject
+}
+
+case class AuthenticatedAuditDetails(channel: ChannelType, enrolmentId: Ior[TURN, EORINumber], json: JsObject) extends AuditDetails {
+
+  def customerId: String =
+    enrolmentId.fold(
+      turn => turn.value,
+      eoriNumber => eoriNumber.value,
+      (_, eoriNumber) => eoriNumber.value
+    )
+
+  def enrolmentType: String =
+    enrolmentId.fold(
+      _ => Constants.LegacyEnrolmentKey,
+      _ => Constants.NewEnrolmentKey,
+      (_, _) => Constants.NewEnrolmentKey
+    )
```

Comment:
I'm not entirely clear what this is doing, can you explain?

Reply:
This is used to add the enrolment type to the audit event body so that we can track users migrating via Splunk. If we have an EORINumber or both an EORINumber and a TURN then we add the new enrolment key, since the user must have signed up for the new enrolment.